torchtune/data/_messages.py

```diff
@@ -209,6 +209,7 @@ def __init__(
             )
 
         self.image_dir = image_dir
+        self.has_system = "system" in self.column_map
```

Comment:
So if a user wants to use a dataset that has a system column, they will need to specify `{"system": "system"}` in column map? It feels a bit unintuitive, ideally they don't need to do this and if there is a system column and it's named system, they shouldn't need to do anything because it follows a default pattern. Still thinking about the best logic for this...